### DISCUSSION TITLE
Enhancement: A button to quickly update the currently displayed library.

### DISCUSSION THREAD

**Discussion by Iturbide** [advplyr/audiobookshelf#2147] [Ideas]
Hello!

Right now, I know of two ways to add a book to a library:

1. Wait for the scheduled scan to run.
2. Settings -> Libraries -> Select library -> ... menu -> scan

But the usual case is: I just added a book and I want it added right now. I have no patience, and now I have to click 5 times!

If you think "This is a paper cut", you are absolutely right.  This is my one and only gripe, so don't take this as criticism of the project. It's just that by now I'm beginning to think I'm not the only one who's annoyed by this, and I wonder how hard it is to put a button somewhere on the front, in the top bar, or wherever you see fit. 

Wouldn't this be a nice enhancement? Or have I missed a completely obvious third method which makes me hang my head in shame?  I'd love to hear other peoples views.

thank you for your time and attention.

**Comment by advplyr** [comment]
Hey, most people use the file watcher that automatically scans items when files are added/removed. When I add a new book it pops up in my library in a few seconds.

The watcher doesn't work for every file system, like network file systems.

**Reply by Iturbide** [reply]
Hello. You are spot on: NFS here. So that one's out.

But let me add some considerations. 
I see Audiobookshelf itself very much as a server application. In docker, it can run on pretty much anything. It's pretty light and is easily tucked away on a Raspberry Pi, like I have done. And how do you serve storage to a Pi? Nfs is the easiest choice, followed by cifs if that's where your data lives.  So using inotify (just guessing here) or similar is very convenient, but limited in scope.  It does what you ask of it, and nothing more. Which is fine.
Now I'm not sure if someone has already come up with an inotify equivalent for networked file systems, but that goes certainly waaay beyond the scope of this enhancement request. 
All I would like to have is a button on the front of the gui. Nada mas. To save me 5 clicks. Just to give you an idea about how urgently I see this. It's not even a feature request, it's an enhancement. 

But closely related to that, my goal would be very well served with a console command to kick off a library scan. I could run that by itself, or tie it into a script of my own making. Any pointers on that? It'd be very helpful.
